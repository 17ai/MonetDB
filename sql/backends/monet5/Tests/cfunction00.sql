create function myaggr()     
returns integer
begin
    declare s int;
	set s = 0;
	while (true)
	do
		set s = s + 1;
		return s ; --yield s;
	end while;
	return s;
END;
select * from functions where name ='myaggr';

-- a continuous procedure can be called like any other procedure
start continuous  myaggr();

select myaggr(); #should return 1
select myaggr(); #should return 2

drop function myproc;

--factory factories.myaggr():int;             	#[0] (0)  0 
--#function user.myaggr():int;             	#[0] (0)  0 
--    As := nil:int;                      	#[1] (0)  2 <- 3 
--    As := 0:int;                        	#[2] (0)  2 <- 5 
--barrier X_8 := true;                    	#[3] (0)  8 <- 7 
--    leave X_8 := false:bit;             	#[4] (0)  8 <- 54 
--    X_12 := As;                         	#[5] (0)  12 <- 2 
--    X_13 := calc.lng(X_12);             	#[6] (0) CMDvarCONVERT 13 <- 12 
--    X_16:lng := calc.+(X_13, 1:lng);    	#[7] (0) CMDvarADDsignal 16 <- 13 15 
--    X_17 := calc.int(X_16);             	#[8] (0) CMDvarCONVERT 17 <- 16 
--    As := X_17;                         	#[9] (0)  2 <- 17 
--	yield myaggr:int := As;
--    # return myaggr:int := As;            	#[10] (0)  0 <- 2 
--    redo X_8 := true;                   	#[11] (0)  8 <- 7 
--exit X_8;                               	#[12] (0)  8 
--    return myaggr:int := As;            	#[13] (0)  0 <- 2 
--end factories.myaggr;                        	#[14] (0)  
