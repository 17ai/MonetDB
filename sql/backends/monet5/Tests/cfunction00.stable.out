stdout of test 'cfunction00` in directory 'sql/backends/monet5` itself:


# 15:54:43 >  
# 15:54:43 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37465" "--set" "mapi_usock=/var/tmp/mtest-14298/.s.monetdb.37465" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/MonetDB-trails/BUILD/var/MonetDB/mTests_sql_backends_monet5"
# 15:54:43 >  

# MonetDB 5 server v11.28.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.498 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2017 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://dhcp-120.eduroam.cwi.nl:37465/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-14298/.s.monetdb.37465
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/Timetrails module loaded

Ready.

# 15:54:43 >  
# 15:54:43 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-14298" "--port=37465"
# 15:54:43 >  

#create function aggr00()     
#returns integer
#begin
#    declare s int;
#	set s = 0;
#	while (true)
#	do
#		set s = s + 1;
#		yield s;
#	end while;
#	return s;
#END;
#select * from functions where name ='aggr00';
% sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions,	sys.functions # table_name
% id,	name,	func,	mod,	language,	type,	side_effect,	varres,	vararg,	schema_id # name
% int,	varchar,	varchar,	varchar,	int,	int,	boolean,	boolean,	boolean,	int # type
% 4,	6,	147,	4,	1,	1,	5,	5,	5,	4 # length
[ 8538,	"aggr00",	"create function aggr00() \nreturns integer\nbegin\n declare s int;\n set s = 0;\n while (true)\n do\n set s = s + 1;\n yield s;\n end while;\n return s;\nend;",	"user",	2,	1,	false,	false,	false,	2000	]
#select aggr00(); #should return 1
% .L2 # table_name
% L2 # name
% int # type
% 1 # length
[ 1	]
#select aggr00(); #should return 2
% .L2 # table_name
% L2 # name
% int # type
% 1 # length
[ 2	]
#drop function aggr00;

# 15:54:44 >  
# 15:54:44 >  "Done."
# 15:54:44 >  

